Sources/ComposableArchitecture/SharedState/PersistenceKey.swift

```diff
@@ -7,10 +7,18 @@
 ///
 /// See the article <doc:SharingState> for more information, in particular the
 /// <doc:SharingState#Custom-persistence> section.
-public protocol PersistenceReaderKey<Value>: Hashable {
```

Comment:
Losing this can be a very slight breaking change to custom persistence strategies, so we could keep the requirement for now, it just isn't useful when using a custom ID type...